Verida Network Grant Application

ysam21 opened the pull request:
### Project Abstract
The Polkadot ecosystem enables a variety of application specific blockchains to provide decentralized computation capabilities. While this is enormously beneficial for building the next generation of Web3 applications, there are additional technical capabilities required to build applications capable of replacing existing Web2 incumbents.

The Verida network provides this missing Web3 infrastructure for developers and end users. Verida will bring single-sign on, decentralized identities, messaging and private data storage to the Polkadot ecosystem.

This is Verida’s first proposal to the W3F Grants program.


### For which grant [level](https://github.com/w3f/Grants-Program#level_slider-levels) are you applying? 
- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [x] **Level 2**:  Up to $50,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for > $100k Web3 Foundation Council approval)

### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied, renamed ( `project_name.md`) and updated.
- [x] A BTC or Ethereum (DAI/USDT) address for the payment of the milestones is provided inside the application.
- [x] I have read and acknowledged the [terms and conditions](https://github.com/w3f/Grants-Program/blob/master/docs/T&Cs.md).
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process).

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/w3f/Grants-Program?pullRequest=887) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/w3f/Grants-Program?pullRequest=887) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/w3f/Grants-Program?pullRequest=887) it.</sub>

Noc2 reviewed (changes requested):
Thanks a lot for the application. I have a few questions/comments: First of all Verida Vault seems to be closed source, we only support open source software development via our grants program. This doesn’t necessarily mean everything needs to be open source, but for example your first milestone seems to be highly focused on closed source development and the deliveries don’t seem necessarily helpful for others in our ecosystem. Is this correct? If this is the case, it might make more sense to apply for treasury funding. Additionally deliverables 0a-0d of the template are usually mandatory for all milestones, and deliverable 0e at least for the last one. See https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md Also you currently apply for a level 2 grant, but you are requesting more than 50k for the entire grant. My recommendation would be initially only to apply for one milestone and we can sign a follow-up grant later. Otherwise, we need to get council approval, which usually takes some time.

ysam21 commented:
Hi @Noc2 , thank you for providing feedback! We've taken all of your points into consideration and have addressed them in the latest revision. To summarise the changes, we have:
- removed development on the Verida Vault (originally Milestone 1) from the grant application. Since Polkadot support in the Vault is necessary to proceed with the next milestone, Verida will complete this work separately as self-funded development.
- removed Milestone 3 so that this initial grant application can stay within the Level 2 amount. We would be keen to revisit this as a follow-up grant.
- added in the remaining 0.x deliverables.

Hopefully these changes address your comments, look forward to hearing back!

Noc2 reviewed (approved):
Thanks for the update. I’m happy to support it and share it with the rest of the team.

ysam21 commented:
Hi @Noc2, does anyone from the team have any questions on the application or need clarifications from us? Happy to set up a call if needed!

Noc2 commented:
Thanks for chasing us. I will share the application again with everyone. Sorry for the delay. The is currently trying to catch-up with everything.

gautamdhameja reviewed (approved):
Looks good to me

semuelle reviewed (commented):
Thank you for this application, @ysam21. It looks very interesting. I only have one question, since you don't want to provide infrastructure: based on the deliverables I assume that everything you are building as part of the grant will be *usable* by the end of the grant. So, on top of unit tests, I will be able to test the functionality by following the documentation on submitting requests and data. Correct?

ysam21 commented:
Thanks for the question @semuelle. Yes absolutely, you'll be able to test using the Verida Vault wallet (which is freely available), developer documentation, and we'll have a number of demo apps to showcase all of the functionality listed in the deliverables.

ysam21 commented:
Thank you @semuelle @Noc2 @gautamdhameja for approving our grant application!
We will first complete the self-funded contribution (consider it Milestone 0) which is a prerequisite for Milestone 1.
We look forward to having Verida part of the Polkadot ecosystem!

alxs commented:
Hey @ysam21, are you still interested in this? When are you planning to submit?

alxs commented:
@ysam21 unless you plan to deliver soon, could you please submit an [amendment](https://github.com/w3f/Grants-Program#changes-to-a-grant-after-approval) with an updated timeline? Note that if we don't hear from you within the next two weeks, we'll assume you're no longer interested and terminate the grant.

ysam21 commented:
Hey @alxs thanks for checking in! I will discuss with engineering on what timeline we're looking at. Cheers

ysam21 commented:
Hi @alxs, the development timeline still holds i.e. once the self-funded contribution is completed, the expected development time is 2 months. We have had a shift in priorities to focus on decentralization of the testnet and user growth, so that we can strengthen the value of what we bring to the Dotsama ecosystem. We intend to deliver this closer to Q2 2023.

alxs commented:
Sounds good @ysam21. In that case, could you please submit an amendment as I mentioned before and include a delivery date? Otherwise, the date the application is accepted [is considered](https://github.com/w3f/Grants-Program#2-application-review) the starting date of the first milestone.

keeganquigley commented:
Hi @ysam21 I believe the amendment specified that a delivery would be submitted in June, are you able to provide an update? Thanks!

keeganquigley commented:
pinging @ysam21

ysam21 commented:
Hi @keeganquigley, thanks for the checking in! We'd like to align our proposal with the challenges faced by dapp developers, based on the insights we've gained from engaging with the community. Are you open for us to update the grant proposal?

keeganquigley commented:
Thanks @ysam21 yes absolutely, feel free to make changes to the original proposal and then submit a pull request. The amendment will simply require 3 approvals from the committee in order to be merged. If necessary, you can extend the timeline as well. Thanks!

keeganquigley commented:
pinging @ysam21 for an update.

semuelle commented:
One last ping before we cancel this grant: @ysam21, @nick-verida, @tahpot
